src/rules/no-unused-collection.ts

```diff
@@ -27,6 +27,9 @@ import { collectionConstructor, writingMethods } from "../utils/collections";
 const message = "Either use this collection's contents or remove the collection.";
 
 const rule: Rule.RuleModule = {
+  meta: {
+    type: "problem",
```

Comment:
should be `suggestion`, on its own unused collection is not a problem (https://jira.sonarsource.com/browse/RSPEC-4030 is a code smell)

Reply:
The above are I think reasonable points, but just to consider, despite the name "problem", ESLint describes a "problem" as "potential errors" ([1](https://eslint.org/docs/user-guide/command-line-interface#fix-type)) or "code that either will cause an error or may cause a confusing behavior." ([2](https://eslint.org/docs/developer-guide/working-with-rules#rule-basics))--not necessarily that they are always problems (there are few rules that are *always* problems, esp. since rules only apply if code can be parsed). The "suggestion" value, on the other hand, suggest that no errors will occur as a result of the issue, i.e., for "something that could be done in a better way but no errors will occur if the code isn't changed." (e.g., like a rule to require that variables are in the [top of the function](https://eslint.org/docs/rules/vars-on-top)).

So if you really have your own internal distinctions you want to map to "problem"/"suggestion", I understand and can change, but in my view, the behavior when the user uses `eslint --fix-type problem` or `eslint --fix-dry-run problem` is to ensure even those which are only "potential" errors get fixed or reported.

Reply:
Ok, you convinced me :)